src/thunderbots/software/util/canvas_messenger/canvas_messenger.h

```diff
@@ -0,0 +1,129 @@
+/**
+ * The Canvas Messenger is a singleton object that receives draw calls
+ * to draw various shapes.
+ *
+ * The singleton constructs a binary representation of all the sprites
+ * to be drawn for a particular layer and sends it to the visualizer via
+ * ROS messages.
+ */
+
+#pragma once
+
+#include <ros/ros.h>
+
+#include <chrono>
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "thunderbots_msgs/CanvasLayer.h"
+#include "util/constants.h"
+
+namespace Util
+{
+    class CanvasMessenger
+    {
+       public:
+        /**
+         * Sprite is a struct that contains all the information
+         * necessary to create a sprite. The default constructor
+         * creates a 100x100 white rectangle sprite.
+         */
+        typedef struct Sprite
+        {
+            Sprite()
+                : texture(0),
+                  x(0.0),
+                  y(0.0),
+                  width(100.0),
+                  height(100.0),
+                  rotation(0.0),
+                  opacity(255),
+                  red(255),
+                  green(255),
+                  blue(255)
+            {
+            }
+
+            uint8_t texture;
+            int16_t x;
+            int16_t y;
+            int16_t width;
+            int16_t height;
+            int16_t rotation;
+            uint8_t opacity;
+            uint8_t red;
+            uint8_t green;
+            uint8_t blue;
+        } Sprite;
+
+       public:
+        /**
+         * Getter of the singleton object.
+         *
+         * @return A shared pointer of the static instance
+         */
+        static std::shared_ptr<CanvasMessenger> getInstance();
+
+        void initializePublisher(ros::NodeHandle node_handle);
+
+        /**
+         * Uses ROS publishers to publish sprite data for each layer and
+         * then clears all layer data.
+         */
+        void publishAndClearLayers();
+
+        /**
+         * Clears all sprite data for all layers
+         */
+        void clearLayers();
+
+        /**
+         * Draw a sprite onto a specific layer.
+         *
+         * @param layer: The layer number this shape is being drawn to
+         * @param sprite: the sprite data to draw
+         */
+        void drawSprite(uint8_t layer, Sprite sprite);
```

Comment:
we should probably declare this guy private since the intent is to (as a next PR) write some abstractions like `drawRectangle`, `drawCircle`, etc., and we want to avoid anyone developing a dependency on it in the meantime.......

Reply:
After discussion with Gareth we can leave this for now in the interest of getting this PR merged. No one should be using this for a few days until it's wrapped up with abstractions.